drv/stm32xx-i2c/src/lib.rs

```diff
@@ -148,33 +159,48 @@ pub enum ReadLength {
     Variable,
 }
 
+#[allow(clippy::upper_case_acronyms)]
 #[derive(Copy, Clone, Eq, PartialEq)]
+enum Register {
+    CR1,
+    CR2,
+    OAR1,
+    OAR2,
+    TIMINGR,
+    TIMEOUTR,
+    ISR,
+    PECR,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, counters::Count)]
 enum Trace {
-    WaitISR(u32),
-    WriteISR(u32),
-    WriteWaitISR(u32),
-    ReadISR(u32),
-    ReadWaitISR(u32),
-    RxISR(u32),
-    KonamiISR(u32),
-    Konami(I2cKonamiCode),
-    ResetISR(u32),
-    ResetCR2(u32),
-    AddrISR(u32),
+    Wait(Register, u32),
+    Write(Register, u32),
+    WriteWait(Register, u32),
+    Read(Register, u32),
+    ReadWait(Register, u32),
+    KonamiOperation(I2cKonamiCode),
+    Konami(Register, u32),
+    Reset(Register, u32),
+    Addr(Register, u32),
     AddrMatch,
     AddrNack(u8),
+    RxReg(Register, u32),
     Rx(u8, u8),
     RxNack(u8, u8),
     Tx(u8, u8),
     TxBogus(u8),
     TxOverrun(u8),
-    TxISR(u32),
+    TxReg(Register, u32),
```

Comment:
and similarly, do we want to add `#[count(children)]` here as well?

Reply:
Similarly, no. ;)